### PR TITLE
Community Pipeline: Add z-image differential img2img

### PR DESCRIPTION
**Pull request by r4inm4ker**
# What does this PR do?

Adds [Differential Diffusion](https://arxiv.org/abs/2306.00950) to [Z-Image](https://github.com/Tongyi-MAI/Z-Image)
Mostly copied from  pipeline_flux_differential_img2img.py implementation.

## Before submitting
- [x] Did you read the [contributor guideline](https://github.com/huggingface/diffusers/blob/main/CONTRIBUTING.md)?
- [x] Did you read our [philosophy doc](https://github.com/huggingface/diffusers/blob/main/PHILOSOPHY.md) (important for complex PRs)?

How to test:
```py
import torch
from pipeline_z_image_differential_img2img import ZImageDifferentialImg2ImgPipeline
from diffusers.utils import load_image

pipe = ZImageDifferentialImg2ImgPipeline.from_pretrained("Z-a-o/Z-Image-Turbo", torch_dtype=torch.bfloat16)
pipe.to("cuda")

init_image = load_image("https://github.com/exx8/differential-diffusion/blob/main/assets/input.jpg?raw=true")

mask = load_image("https://github.com/exx8/differential-diffusion/blob/main/assets/map.jpg?raw=true")

prompt = "painting of a mountain landscape with a meadow and a forest, meadow background, anime countryside landscape"

image = pipe(
 prompt,
 image=init_image,
 mask_image=mask,
 strength=0.75,
 num_inference_steps=9,
 guidance_scale=0.0,
 generator=torch.Generator("cuda").manual_seed(0),
).images[0]
image.save("image.png")
```

## Who can review?
@yiyixuxu @asomoza

### CODE REVIEW

**Comment by asomoza** [comment]
@bot /style

**Comment by asomoza** [comment]
@r4inm4ker thanks a lot! the bot can't fix the style, can you please run:

```bash
make style
make quality
```

also if it doesn't fix it, there's some white spaces in some empty lines in the example doc string

**Comment by asomoza** [comment]
in the meantime, this is the apple pear test with this one:

|original|mask|result|
|---|---|---|
||gradient_mask|zimage_diff_diff|

this one is more finicky too than SDXL, I had to up the strength to 0.9 and use 12 steps to get that result

**Comment by bghira** [comment]
need sigma shift

**Comment by r4inm4ker** [comment]
Thanks for the feedback ! I will work on the updates after coming back from holiday.

**Comment by r4inm4ker** [comment]
Hi all, Happy New Year !
I have run the make style and make quality and pushed an update.

@bghira Regarding the "need sigma shift" comment : I am not familiar with what it means as I am still very new to ML/diffusion and this is my first attempt to contribute to the community, so any help to elaborate or explain a bit more would be really appreciated. Thanks !

In the meanwhile, I have tried rerunning example inputs from https://differential-diffusion.github.io/ , and this is the result with strength 1.0 and steps 9. prompt : ""tree of life under the sea"

| original| mask | result
|----------|----------|----------|
|  |  |  |

**Comment by asomoza** [comment]
@r4inm4ker the sigma shift comment was for me I think. If I make the shift higher I don't need to use a higher strength or more steps.

Your example works fine because you're using a strength of 1.0, I don't use it like that since I use diff-diff more for inpainting.

But since this is a community pipeline, it just needs to work and pass the basic tests, I do see some code that can be improved but it's not necessary at this point.

**Comment by asomoza** [comment]
@r4inm4ker thanks a lot! failing tests are not related to this PR

**Comment by r4inm4ker** [comment]
Thanks for merging !
